src/node/stm32_hardware.h

```diff
@@ -65,33 +65,37 @@ class Stm32Hardware : public Hardware {
         }
     }
 
+#ifdef ENABLE_SERIALUSB
+    void initUSBSerial(USBSerial& ser, unsigned long baudRate) {
+        ser.begin(baudRate);  // Start serial interface
+        int count = 0;
+        while (!ser || count++ < 1000) {
+            delay(1);  // Wait for the Serial port to initialize
+        }

```

Comment:
I would also add a comment here:
// ser may not be initialized
so ppl dont get caught unaware.
Or better, have the function return bool, true for initialized, false for not so it is clear there are two final states.
Given ser maybe uninit, we need to guard against using it, so stm32_hardware.cpp handlestreamevent should be guarded by an if stmt.
hardware.h initSerial() should be changed similarly and serialEvent() in other *_hardware.cpp.